homeassistant/components/enocean/switch.py

```diff
@@ -25,18 +27,48 @@
 )
 
 
-def setup_platform(
+def generate_unique_id(dev_id: list[int], channel: int) -> str:
+    """Generate a valid unique id."""
+    return f"{combine_hex(dev_id)}{channel}"
```

Comment:
We separate parts with dash in the sensor unique ids.
```suggestion
    return f"{combine_hex(dev_id)}-{channel}"
```